### PR TITLE
installer: Add Azure support

### PR DESCRIPTION
**Pull request by jvatic**

### CODE REVIEW

**Comment by jvatic** [comment]
ping @titanous

File: installer/app/src/views/asset-paths.js.erb [hunk 1]
```diff
@@ -19,6 +19,20 @@ var AssetPaths = <%= Yajl::Encoder.encode([
 
 	"aws-logo.png",
 	"digitalocean-logo.png",
+	"azure-logo.png",
+
+  "azure-app-0.png",
```
Comment (titanous) [inline installer/app/src/views/asset-paths.js.erb]: Indentation

File: installer/app/src/views/azure-create-app-tutorial.js.jsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+					<p>Click the "Windows Azure Service ..." option</p>
+					<p>Click the checkmark in the bottom right to continue</p>
+				</li>
+
+				<li>
+					<img src={AssetPaths["azure-app-10.png"]}  style={{
+						width: 659,
+						height: 169
+					}}/>
+					<p>Click the "Deligated Permissions" dropdown</p>
```
Comment (titanous) [inline installer/app/src/views/azure-create-app-tutorial.js.jsx]: Delegated

File: installer/azure_template.go [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+    "NetworkSecurityGroupName": "[concat(parameters('ClusterID'), 'sg')]",
+    "SubnetName": "[concat(parameters('ClusterID'), 'subnet')]",
+    "SubnetRef": "[concat(variables('VirtualNetworkID'), '/subnets/', variables('SubnetName'))]",
+    "NetworkInterfaceName": "[concat(parameters('ClusterID'), 'nic')]",
+    "AddressPrefix": "10.0.0.0/16",
+    "SubnetPrefix": "10.0.0.0/24",
+    "VirtualMachineName": "[concat(parameters('ClusterID'), 'vm')]",
+    "AuthorizedKeysPath": "[concat('/home/', parameters('VirtualMachineUser'), '/.ssh/authorized_keys')]"
+  },
+	"resources": [
```
Comment (titanous) [inline installer/azure_template.go]: Indentation

File: installer/azure_template.go [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+              "sourcePortRange": "*",
+              "destinationPortRange": "2222",
+              "sourceAddressPrefix": "*",
+              "destinationAddressPrefix": "*",
+              "access": "Allow",
+              "priority": 104,
+              "direction": "Inbound"
+            }
+          }
+        ]
```
Comment (titanous) [inline installer/azure_template.go]: Missing the user tcp port range (3000-3500)

File: installer/schema.go [hunk 5]
```diff
@@ -23,11 +24,21 @@ type Cluster interface {
...
+	ID         string             `json:"id" ql:"index xID"`
+	Secret     string             `json:"secret"`
+	Name       string             `json:"name"`
+	Type       string             `json:"type"` // enum(aws, digital_ocean, azure)
+	OAuthCreds []*OAuthCredential `json:"oauth_creds,omitempty" ql:"-"`
+	DeletedAt  *time.Time         `json:"deleted_at,omitempty"`
+}
+
+type OAuthCredential struct {
+	ClientID     string     `json:"client_id"` // ClientID <-> Credential.ID
```
Comment (titanous) [inline installer/schema.go]: Not sure what the `<->` means?

Comment (jvatic) [inline installer/schema.go]: Credential.ID and OAuthCredential.ClientID are the same, and `==` would probably be better

**Comment by jvatic** [comment]
@titanous comments addressed

**Comment by jvatic** [comment]
There's an intermittent issue with the OAuth flow (token response is sometimes a 400 (AADSTS65001)), we're waiting to hear back from the Azure team.

**Comment by felixrieseberg** [comment]
@dushyantgill is one of the Senior PMs on Azure's identify flows, I'm waiting to hear back from him - he's been looking into similar issues.

**Comment by jvatic** [comment]
Thanks @felixrieseberg for all your help!

@titanous This should be good to go now

**Comment by titanous** [comment]
LGTM

**Pull request merged**